lib/main.dart

```diff
@@ -216,7 +216,12 @@ class _MyHomePageState extends State<MyHomePage> {
   }
 }
 
+// Maybe add input validation here, it is true that when you add a task with empty fields you cannot do it
+// but it would be better to show a message to the user.
 class PendingTasksPage extends StatelessWidget {
+
+  //  to truly reduce the scope of these variables, you can encapsulate them within a custom widget or class
```

Comment:
This is not truly necessary but in my opinion it would make sense and code would be more redable.